src/cli/commands/check.js

```diff
@@ -336,7 +337,20 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
           const foundHuman = `${humaniseLocation(path.dirname(depPkgLoc)).join('#')}@${depPkg.version}`;
           if (!semver.satisfies(depPkg.version, range, config.looseSemver)) {
             // module isn't correct semver
-            reportError('packageDontSatisfy', subHuman, foundHuman);
+            const resRange = mainPackageJson.resolutions && mainPackageJson.resolutions[name];
+            if (resRange) {
+              const resHuman = `${human}#${name}@${resRange}`;
+
+              if (semver.satisfies(depPkg.version, resRange, config.looseSemver)) {
+                reporter.warn(reporter.lang('packageDontSatisfy', subHuman, foundHuman));
```

Comment:
From the test case above, I find the warning above a little confusing. Would you mind using [incompatibleResolutionVersion](https://github.com/yarnpkg/yarn/blob/master/src/reporters/lang/en.js#L126) instead? 

Currently it reads `warning.*"pad-left#repeat-string@\\^1.5.4" doesn't satisfy found match of "repeat-string@1.4.0"` 

but it'd be better to say

`warning.*Resolution field (repeat-string@1.4.0) is incompatible with requested version (pad-left#repeat-string@\\^1.5.4)'`

Reply:
👍
done.

rebased onto current master once again